src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder.cs

```diff
@@ -101,6 +101,7 @@ internal override void BindPatternSwitchLabelForInference(CasePatternSwitchLabel
         /// the decision tree.
         /// </summary>
         private ImmutableArray<BoundPatternSwitchSection> BindPatternSwitchSections(
+            SyntaxNode syntax,
```

Comment:
>SyntaxNode syntax [](start = 12, length = 17)

It looks like there is only one call-site. Consider making this a SwitchStatementSyntax and removing the two next parameters (boundSwitchExpression is coming from a property, sections are available from SwitchStatementSyntax).